src/wazuh_db/wdb_upgrade.c

```diff
@@ -300,16 +296,14 @@ int wdb_adjust_upgrade(wdb_t *wdb, int upgrade_step) {
 int wdb_adjust_global_upgrade(wdb_t *wdb, int upgrade_step) {
     switch (upgrade_step) {
         case 3:
-            return wdb_global_adjust_v4(wdb);
+            // Migrating to the fourth version of the database: The groups_hash
+            // column is calculated with the hash of the group column
+            return wdb_global_update_all_agents_groups_hash(wdb);
```

Comment:
Honestly, I prefer the old version of having **wdb_global_adjust_v4**, keeping in mind that this **switch** could grow with each DB version, new versions can have many actions that need to be wrapped into one method, and will be easier to read if we have all the scenarios into a method and don´t mix them.
But, this could be discussed.

Reply:
As we discussed, I evaluated the next alternatives:

- Moving the `wdb_global_update_all_agents_groups_hash` function from the `wdb_global.c` file to `wdb_upgrade.c` would probably be the best approach. The thing is that it would require moving also the `wdb_global_update_agent_groups_hash` function and the unit test cases of both. So we concluded that this change would be expensive in terms of effort.
- Restoring the logic as it was before this pull request was not a good practice because we had a function just calling another function.
- Considering the previous alternatives, I decided to rename `wdb_global_update_all_agents_groups_hash` to `wdb_global_adjust_v4` and remove the comment from the switch case.